tt_metal/common/test_tiles.hpp

```diff
@@ -10,50 +10,61 @@
 
 #include <cstdint>
 #include <vector>
+#include <optional>
+#include "tt_metal/common/constants.hpp"
 #include "tt_metal/common/assert.hpp"
 #include "tt_metal/third_party/tracy/public/tracy/Tracy.hpp"
 #include "math.hpp"
 
 enum TensorLayout {
     LIN_ROW_MAJOR = 0, // standard element-wise row-major
-    TILED32_SWIZZLED = 1, // row-major of tiles 32x32, each tile is row-major-swizzled
-    TILED32_4FACES = 2,  // rowm major of tiles 32x32, each tile is 4 faces, each face is row-major, faces are swizzled
+    TILED_SWIZZLED = 1, // row-major of tiles, each tile is row-major-swizzled
+    TILED_NFACES = 2,  // row-major of tiles, each tile is N (N = 1, 2, or 4) faces, each face is row-major, faces are swizzled
 };
 
 template <class T, template <typename...> typename BufferType>
-std::vector<T> convert_to_tile_layout(const BufferType<T>& data) {
+std::vector<T> convert_to_tile_layout(
+    const BufferType<T>& data,
+    const std::optional<std::vector<uint32_t>>& tile_shape = std::nullopt,
+    const std::optional<const std::vector<uint32_t>>& face_shape = std::nullopt) {
     ZoneScoped;
     std::vector<T> result;
```

Comment:
You need to reserve result to avoid relocations. But I do not know how much perf matters here.

Reply:
changed